make/catalog.mk

```diff
@@ -0,0 +1,52 @@
+##@ Operator Catalog
+
+# The image tag given to the resulting catalog image (e.g. make catalog-build CATALOG_IMG=example.com/operator-catalog:v0.2.0).
+CATALOG_IMG ?= $(IMAGE_TAG_BASE)-catalog:$(IMAGE_TAG)
+
+CATALOG_FILE = $(PROJECT_PATH)/catalog/authorino-operator-catalog/operator.yaml
+CATALOG_DOCKERFILE = $(PROJECT_PATH)/catalog/authorino-operator-catalog.Dockerfile
+
+$(CATALOG_DOCKERFILE): $(OPM)
+	-mkdir -p $(PROJECT_PATH)/catalog/authorino-operator-catalog
+	cd $(PROJECT_PATH)/catalog && $(OPM) generate dockerfile authorino-operator-catalog
+catalog-dockerfile: $(CATALOG_DOCKERFILE) ## Generate catalog dockerfile.
+
+$(CATALOG_FILE): $(OPM) $(YQ)
+	@echo "************************************************************"
+	@echo Build authorino operator catalog
+	@echo
+	@echo BUNDLE_IMG					= $(BUNDLE_IMG)
+	@echo CHANNELS						= $(CHANNELS)
+	@echo "************************************************************"
+	@echo
+	@echo Please check this matches your expectations and override variables if needed.
+	@echo
+	$(PROJECT_PATH)/utils/generate-catalog.sh $(OPM) $(YQ) $(BUNDLE_IMG) $@ $(CHANNELS)
+
+.PHONY: catalog
+catalog: $(OPM) ## Generate catalog content and validate.
+	# Initializing the Catalog
+	-rm -rf $(PROJECT_PATH)/catalog/authorino-operator-catalog
+	-rm -rf $(PROJECT_PATH)/catalog/authorino-operator-catalog.Dockerfile
+	$(MAKE) $(CATALOG_DOCKERFILE)
+	$(MAKE) $(CATALOG_FILE) BUNDLE_IMG=$(BUNDLE_IMG)
+	cd $(PROJECT_PATH)/catalog && $(OPM) validate authorino-operator-catalog
+
+# Build a catalog image by adding bundle images to an empty catalog using the operator package manager tool, 'opm'.
+# Ref https://olm.operatorframework.io/docs/tasks/creating-a-catalog/#catalog-creation-with-raw-file-based-catalogs
+.PHONY: catalog-build
+catalog-build: ## Build a catalog image.
+	# Build the Catalog
+	docker build $(PROJECT_PATH)/catalog -f $(PROJECT_PATH)/catalog/authorino-operator-catalog.Dockerfile -t $(CATALOG_IMG)
+
+# Push the catalog image.
+.PHONY: catalog-push
+catalog-push: ## Push a catalog image.
+	$(MAKE) docker-push IMG=$(CATALOG_IMG)
+
+deploy-catalog: $(KUSTOMIZE) $(YQ) ## Deploy operator to the K8s cluster specified in ~/.kube/config using OLM catalog image.
```

Comment:
nit

Error raised when `deploy-catalog` is run on a repo without `bin/kustomize`

```
make: *** No rule to make target '/home/eguzki/git/kuadrant/authorino-operator/bin/kustomize', needed by 'deploy-catalog'.  Stop.
```

you need to add

```make
$(KUSTOMIZE):                                                                      
    $(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v4@v4.5.5)
```

Reply:
https://github.com/Kuadrant/authorino-operator/blob/f4ea39aad23e9980a3f23c4f287818c610cf7d26/Makefile#L113